File: OctopusStore/ClientApp/src/app/view-models/item-variant-characteristic-value/item-variant-characteristic-value.ts
import { Entity } from "../entity";

export class ItemVariantCharacteristicValue extends Entity {
  itemVariantId: number;
  characteristicValueId: number;

  public constructor(init?: Partial<ItemVariantCharacteristicValue>) {
    super(init);
    Object.assign(this, init);
  }
}

File: ApplicationCore/Entities/ItemVariantCharacteristicValue.cs
namespace ApplicationCore.Entities
{
    /// <summary>
    /// Property Value of and Item Variant. For example 32GB of Storage in iPhone variant.
    /// </summary>
    public class ItemVariantCharacteristicValue: Entity
    {
        public int ItemVariantId { get; set; }
        public int CharacteristicValueId { get; set; }
        public CharacteristicValue CharacteristicValue { get; set; }
        public ItemVariant ItemVariant { get; set; }

        public ItemVariantCharacteristicValue()
        {
        }
        public ItemVariantCharacteristicValue(int itemVariantId, int categoryPropertyValueId)
        {
            ItemVariantId = itemVariantId;
            CharacteristicValueId = categoryPropertyValueId;
        }
    }
}
